packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx

```diff
@@ -0,0 +1,627 @@
+          >
+            {convertToLocaleString(calloutProps!.hoverXValue, this.props.culture)}
+          </div>
+        </div>
+        <div
+          className={this._classNames.calloutInfoContainer}
+          style={yValueHoverSubCountsExists ? { display: 'flex' } : {}}
+        >
+          {calloutProps!.YValueHover &&
+            calloutProps!.YValueHover.map((yValue: IYValueHover, index: number, yValues: IYValueHover[]) => {
+              const isLast: boolean = index + 1 === yValues.length;
+              const { shouldDrawBorderBottom = false } = yValue;
+              return (
+                <div
+                  {...getAccessibleDataObject(yValue.callOutAccessibilityData, 'text', false)}
+                  key={`callout-content-${index}`}
+                  style={
+                    yValueHoverSubCountsExists
+                      ? {
+                          display: 'inline-block',
+                          ...(shouldDrawBorderBottom && {
+                            borderBottom: `1px solid ${this.props.theme!.semanticColors.menuDivider}`,
+                            paddingBottom: '10px',
+                          }),
+                        }
+                      : {
+                          ...(shouldDrawBorderBottom && {
+                            borderBottom: `1px solid ${this.props.theme!.semanticColors.menuDivider}`,
+                            paddingBottom: '10px',
+                          }),
+                        }
+                  }
+                >
+                  {this._getCalloutContent(yValue, index, yValueHoverSubCountsExists, isLast)}
+                </div>
+              );
+            })}
+          {!!calloutProps.descriptionMessage && (
+            <div className={this._classNames.descriptionMessage}>{calloutProps.descriptionMessage}</div>
+          )}
+        </div>
+      </div>
+    );
+  };
+
+  private _yValueHoverSubCountsExists(yValueHover?: IYValueHover[]) {
+    if (yValueHover) {
+      return yValueHover.some(
+        (yValue: {
+          legend?: string;
+          y?: number;
+          color?: string;
+          yAxisCalloutData?: string | { [id: string]: number };
+        }) => yValue.yAxisCalloutData && typeof yValue.yAxisCalloutData !== 'string',
+      );
+    }
+    return false;
+  }
+
+  private _getCalloutContent(
+    xValue: IYValueHover,
+    index: number,
+    yValueHoverSubCountsExists: boolean,
+    isLast: boolean,
+  ): React.ReactNode {
+    const marginStyle: React.CSSProperties = isLast ? {} : { marginRight: '16px' };
+    const toDrawShape = xValue.index !== undefined && xValue.index !== -1;
+    const _classNames = getClassNames(this.props.styles!, {
+      theme: this.props.theme!,
+      lineColor: xValue.color,
+      toDrawShape,
+    });
+
+    const { culture } = this.props;
+    const yValue = convertToLocaleString(xValue.y, culture);
+    if (!xValue.yAxisCalloutData || typeof xValue.yAxisCalloutData === 'string') {
```

Comment:
when can y axis callout data be a string?

Reply:
should the expression be &&?

Reply:
Didn't explore. Copied code